### PR TITLE
OAK-10001: Bump up minimal Java version to 11

### PR DESCRIPTION
**Pull request by reschke**

### CODE REVIEW

File: oak-parent/pom.xml [hunk 1]
```diff
@@ -111,25 +110,17 @@
       <plugins>
         <plugin>
           <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.8.1</version>
           <configuration>
+            <target>${java.version}</target>
+            <source>${java.version}</source>
+            <release>${java.version}</release>
```
Comment (kwin) [inline oak-parent/pom.xml]: Rather overwrite `maven.compiler.target` (from ASF parent: https://github.com/apache/maven-apache-parent/blob/5faa5c33e3035e5c83af1ed40b4abf6b131a1149/pom.xml#L95) and set `maven.compiler.release` as well to the same value, as that also affects m-javadoc-p, ...

Comment (kwin) [reply oak-parent/pom.xml]: Also you should overwrite `java.minimalBuildVersion` (https://github.com/apache/maven-apache-parent/blob/5faa5c33e3035e5c83af1ed40b4abf6b131a1149/pom.xml#L93)

Comment (reschke) [reply oak-parent/pom.xml]: Please add to the PR...

File: oak-parent/pom.xml [hunk 2]
```diff
@@ -69,8 +68,11 @@
     <testcontainers.version>1.17.5</testcontainers.version>
     <pax-exam.version>4.13.1</pax-exam.version>
     <groovy.version>2.5.20</groovy.version>
-    <java.version>1.8</java.version>
-    <java.version.signature>java18</java.version.signature>
+
+    <java.version>11</java.version>
+    <maven.compiler.release>${java.version}</maven.compiler.release>
+    <maven.compiler.source>${java.version}</maven.compiler.source>
```
Comment (kwin) [inline oak-parent/pom.xml]: redundant (set in asf parent)

File: oak-parent/pom.xml [hunk 3]
```diff
@@ -111,25 +113,14 @@
       <plugins>
         <plugin>
           <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.8.1</version>
```
Comment (kwin) [inline oak-parent/pom.xml]: why downgrading the newer version set in ASF parent?

**Review by kwin — CHANGES_REQUESTED** [review]

File: oak-parent/pom.xml [hunk 4]
```diff
@@ -69,8 +68,11 @@
     <testcontainers.version>1.17.5</testcontainers.version>
     <pax-exam.version>4.13.1</pax-exam.version>
     <groovy.version>2.5.20</groovy.version>
-    <java.version>1.8</java.version>
-    <java.version.signature>java18</java.version.signature>
+
+    <java.version>11</java.version>
```
Comment (kwin) [inline oak-parent/pom.xml:72]:
```suggestion
    <java.version>11</java.version>
    <java.minimalBuildVersion>${java.version}</java.minimalBuildVersion>
```

File: oak-parent/pom.xml (hunk 2 above)
Comment (reschke) [reply oak-parent/pom.xml]: which part exactly is redundant?

Comment (kwin) [reply oak-parent/pom.xml]: https://github.com/apache/maven-apache-parent/blob/5faa5c33e3035e5c83af1ed40b4abf6b131a1149/pom.xml#L94

File: oak-parent/pom.xml (hunk 3 above)
Comment (reschke) [reply oak-parent/pom.xml]: no idea how that got there; maybe a merge issue

File: oak-parent/pom.xml (hunk 2 above)
Comment (kwin) [reply oak-parent/pom.xml]: Just remove line 74

**Review by kwin — APPROVED** [review]

**Review by anchela — APPROVED** [review]
@reschke , looks good to me at a first glance. but i have to admit that i can't tell what the impact of the removal of the animal-sniffer plugins is.

**Comment by reschke** [comment]
I believe this was needed to check that when targeting Java X, we must ensure not to use APIs from X + 1. AFAIU, this is now taken care by javac itself. @kwin - is that right?

**Comment by kwin** [comment]
Right, look at https://openjdk.org/jeps/247 for details. The release parameter triggers the check.

**Pull request merged**